libraries/chain/controller.cpp

```diff
@@ -706,7 +706,7 @@ struct controller_impl {
 
       protocol_features.init( db );
 
-      if (auto dm_logger = get_deep_mind_logger()) {
+      if (auto dm_logger = get_deep_mind_logger(false)) {
```

Comment:
We don't want to log dry-run transactions. I know `on_startup` does not log anything, but seems cleaner to never call dm_logger if trx is transient.

Reply:
At startup, we don't know transaction types.

Reply:
Sorry, I thought I was looking at transaction_context here.